src/rollbacks.rs

```diff
@@ -14,12 +14,25 @@ pub struct RollbackableHashMap<K: Clone + Hash, V: Clone> {
     pub map: HashMap<K, V>,
 }
 
-impl<K: Clone + Hash, V: Clone> RollbackableHashMap<K, V> {
+impl<K: Clone + Hash + Eq, V: Clone> RollbackableHashMap<K, V> {
     pub fn new() -> Self {
         Self {
             map: HashMap::new(),
         }
     }
+
+    pub fn get_logs_after_snapshot(
+        &self,
+        snapshot: <RollbackableHashMap<K, V> as Rollbackable>::Snapshot,
+    ) -> HashMap<K, (Option<V>, V)> {
```

Comment:
It would be the tuple this function returns, where the first value would be the value in the snapshot and the second the value of the current map. The first one is marked as optional because the key could have been a recent write and so it won't be contained in the snapshot. 

This is not used in the era_vm itself but in the `zksync-era` integration. We could return a struct to make it more declarative, the problem is that it would require a new pr on the `zksync-era` repo. Though, we could annotate this for a future refactor.